spikeinterface/core/frameslicerecording.py

```diff
@@ -0,0 +1,60 @@
+import numpy as np
+
+from .baserecording import BaseRecording, BaseRecordingSegment
+
+class FrameSliceRecording(BaseRecording):
+    """
+    Class to get a lazy frame slice.
+    Work only with mono segment recording.
+    """
+
+    def __init__(self, parent_recording, start_frame=None, end_frame=None):
+        channel_ids = parent_recording.get_channel_ids()
+    
+        assert parent_recording.get_num_segments() == 1, 'FrameSliceRecording work only with one segment'
+        
+        parent_size = parent_recording.get_num_samples(0)
+        if start_frame is None:
+            start_frame = 0
+        else:
+            assert 0 <= start_frame < parent_size
+
+        if end_frame is None:
+            end_frame = parent_size
+        else:
+            assert 0 <= start_frame <= parent_size
+
+        BaseRecording.__init__(self,
+                               sampling_frequency=parent_recording.get_sampling_frequency(),
+                               channel_ids=channel_ids,
+                               dtype=parent_recording.get_dtype())
+
+        # link recording segment
+        parent_segment = parent_recording._recording_segments[0]
+        sub_segment = FrameSliceRecordingSegment(parent_segment, start_frame, end_frame)
+        self.add_recording_segment(sub_segment)
+
+        # update dump dict
+        self._kwargs = {'parent_recording': parent_recording.to_dict(), 'start_frame': int(start_frame), 'end_frame': int(end_frame)}
+
+
+class FrameSliceRecordingSegment(BaseRecordingSegment):
+    def __init__(self, parent_recording_segment, start_frame, end_frame):
+        BaseRecordingSegment.__init__(self)
+        self._parent_recording_segment = parent_recording_segment
+        self.start_frame = start_frame
+        self.end_frame = end_frame
+        
+
+    def get_num_samples(self):
+        return self.end_frame - self.start_frame
+
+    def get_traces(self, start_frame, end_frame, channel_indices):
+        if start_frame is None:
+            start_frame = 0
+        if end_frame is None:
+            end_frame = self.get_num_samples()
```

Comment:
@samuelgarcia can you move the parsing of `start_frame` and `end_frame` and also casting to int in the `BaseRecording.get_traces()`?

Reply:
I prefer not.